### ISSUE TITLE
ConnectHardwareKeyboard set to false by default

### ISSUE THREAD

**Issue by userException** [fastlane/fastlane#16083]
I see that this has been fixed here: https://github.com/fastlane/fastlane/pull/12829

Is it available in the latest fastlane 2.142.0 ?

I am facing this issue and I know its because of ConnectHardwareKeyboard only because my XCode UI test also fails when "ConnectHardwareKeyboard" is set to yes.

**Comment by janpio** [comment]
https://github.com/fastlane/fastlane/pull/12829#issuecomment-406388973

**Comment by userException** [comment]
> #12829 (comment)

I have been trying to get this to work but no luck. I had to write a run script in my xcode project to make it work from XCode:
#--------
killall "Simulator"
defaults write com.apple.iphonesimulator ConnectHardwareKeyboard -bool false
#--------

But even this script doesn't work when I run the snapshot tool. Give me some time, I'll make a demo project and provide the link.

**Comment by userException** [comment]
Please use this repo: https://github.com/userException/TestFastlaneKeyboardIssue

Just run fastlane snapshot. Here is the whole log of error which I am seeing:

[12:39:21]: fastlane detected a Gemfile in the current directory
[12:39:21]: However, it seems like you didn't use `bundle exec`
[12:39:21]: To launch fastlane faster, please use
[12:39:21]: 
[12:39:21]: $ bundle exec fastlane snapshot
[12:39:21]: 
[12:39:21]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[12:39:22]: Successfully loaded '/Users/nipunrajput/Desktop/TestFastlaneKeyboardIssue/fastlane/Snapfile' 📄

+----------------------------+----------------------------------+
|          Detected Values from './fastlane/Snapfile'           |
+----------------------------+----------------------------------+
| devices                    | ["iPhone X"]                     |
| languages                  | ["en-US"]                        |
| scheme                     | TestFastlaneKeyboardIssueUITests |
| output_directory           | ./screenshots                    |
| clear_previous_screenshots | true                             |
+----------------------------+----------------------------------+

[12:39:22]: Resolving Swift Package Manager dependencies...
[12:39:22]: $ xcodebuild -resolvePackageDependencies -scheme TestFastlaneKeyboardIssueUITests -project ./TestFastlaneKeyboardIssue.xcodeproj
[12:39:23]: ▸ resolved source packages: 
[12:39:23]: $ xcodebuild -showBuildSettings -scheme TestFastlaneKeyboardIssueUITests -project ./TestFastlaneKeyboardIssue.xcodeproj

+----------------------------+------------------------------------------------------------------+
|                                 Summary for snapshot 2.142.0                                  |
+----------------------------+------------------------------------------------------------------+
| project                    | ./TestFastlaneKeyboardIssue.xcodeproj                            |
| output_directory           | /Users/nipunrajput/Desktop/TestFastlaneKeyboardIssue/screenshots |
| devices                    | ["iPhone X"]                                                     |
| languages                  | ["en-US"]                                                        |
| launch_arguments           | [""]                                                             |
| output_simulator_logs      | false                                                            |
| skip_open_summary          | false                                                            |
| skip_helper_version_check  | false                                                            |
| clear_previous_screenshots | true                                                             |
| reinstall_app              | false                                                            |
| erase_simulator            | false                                                            |
| localize_simulator         | false                                                            |
| buildlog_path              | ~/Library/Logs/snapshot                                          |
| clean                      | false                                                            |
| scheme                     | TestFastlaneKeyboardIssueUITests                                 |
| number_of_retries          | 1                                                                |
| stop_after_first_error     | false                                                            |
| result_bundle              | false                                                            |
| concurrent_simulators      | true                                                             |
| disable_slide_to_type      | false                                                            |
| xcode_path                 | /Applications/Xcode.app                                          |
+----------------------------+------------------------------------------------------------------+

[12:39:25]: Clearing previously generated screenshots
[12:39:25]: Building and running project - this might take some time...
[12:39:25]: Patching simulators '/Users/nipunrajput/Library/Preferences/com.apple.iphonesimulator.plist' to scale to 100%
[12:39:28]: Using device named 'iPhone X' with version '13.3' because no match was found for version '13.2'
[12:39:28]: Running snapshot on: iPhone X
[12:39:28]: $ set -o pipefail && xcodebuild -scheme TestFastlaneKeyboardIssueUITests -project ./TestFastlaneKeyboardIssue.xcodeproj -derivedDataPath '/var/folders/8q/5x83djvd1x111hxf7q7r6pp00000gn/T/snapshot_derived20200307-90012-1ktrb5p' -destination 'platform=iOS Simulator,name=iPhone X,OS=13.3' FASTLANE_SNAPSHOT=YES build test | tee /Users/nipunrajput/Library/Logs/snapshot/TestFastlaneKeyboardIssueUITests-TestFastlaneKeyboardIssueUITests.log | xcpretty 
[12:39:28]: ▸ Loading...
[12:39:29]: ▸ Compiling ViewController.swift
[12:39:34]: ▸ Compiling AppDelegate.swift
[12:39:34]: ▸ Linking TestFastlaneKeyboardIssue
[12:39:35]: ▸ Compiling TestFastlaneKeyboardIssueUITests.swift
[12:39:35]: ▸ Compiling SnapshotHelper.swift
[12:39:36]: ▸ Compiling Main.storyboard
[12:39:36]: ▸ Compiling LaunchScreen.storyboard
[12:39:36]: ▸ Processing Info.plist
[12:39:36]: Running Tests: ▸ Touching TestFastlaneKeyboardIssue.app (in target 'TestFastlaneKeyboardIssue' from project 'TestFastlaneKeyboardIssue')
[12:39:36]: ▸ Processing Info.plist
[12:39:36]: ▸ Linking TestFastlaneKeyboardIssueUITests
[12:39:37]: ▸ Copying /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/lib/libXCTestSwiftSupport.dylib
[12:39:37]: ▸ Copying /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks/XCTest.framework
[12:39:37]: ▸ Copying /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/PrivateFrameworks/XCTAutomationSupport.framework
[12:39:37]: Running Tests: ▸ Touching TestFastlaneKeyboardIssueUITests.xctest (in target 'TestFastlaneKeyboardIssueUITests' from project 'TestFastlaneKeyboardIssue')
[12:39:37]: ▸ Build Succeeded
[12:39:37]: ▸ Linking TestFastlaneKeyboardIssue
[12:39:37]: ▸ Compiling TestFastlaneKeyboardIssueUITests.swift
[12:39:37]: ▸ Compiling SnapshotHelper.swift
[12:39:38]: ▸ Linking TestFastlaneKeyboardIssueUITests
[12:39:39]: ▸ 2020-03-07 12:39:39.056 xcodebuild[90166:2911150]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:
[12:39:39]: ▸ /var/folders/8q/5x83djvd1x111hxf7q7r6pp00000gn/T/snapshot_derived20200307-90012-1ktrb5p/Logs/Test/Run-TestFastlaneKeyboardIssueUITests-2020.03.07_12-39-29-+0530.xcresult/Staging/2_Test/Diagnostics/TestFastlaneKeyboardIssueUITests-BF9B135C-7F22-4683-B258-29285797CF47/TestFastlaneKeyboardIssueUITests-ABA9EECC-E175-4AD3-B375-01B66DDDEA93/Session-TestFastlaneKeyboardIssueUITests-2020-03-07_123939-Dgy80z.log
[12:39:39]: ▸ 2020-03-07 12:39:39.057 xcodebuild[90166:2910066] [MT] IDETestOperationsObserverDebug: (8F78A621-AB00-43E4-AAE7-939C0E275A1F) Beginning test session TestFastlaneKeyboardIssueUITests-8F78A621-AB00-43E4-AAE7-939C0E275A1F at 2020-03-07 12:39:39.057 with Xcode 11C29 on target <DVTiPhoneSimulator: 0x7f85c5a2f740> {
[12:39:39]: ▸ 		SimDevice: iPhone X (207850A5-B6CE-421F-86A8-63EEA7DBF6D2, iOS 13.3, Shutdown)
[12:39:39]: ▸ } (13.3 (17C45))
[12:39:41]: ▸ 2020-03-07 12:39:41.074 xcodebuild[90166:2910066] [MT] IDETestOperationsObserverDebug: (8F78A621-AB00-43E4-AAE7-939C0E275A1F) Finished requesting crash reports. Continuing with testing.
[12:39:46]: ▸ All tests
[12:39:46]: ▸ Test Suite TestFastlaneKeyboardIssueUITests.xctest started
[12:39:46]: ▸ TestFastlaneKeyboardIssueUITests
[12:39:53]: ▸     ✗ testExample, Failed to synthesize event: Neither element nor any descendant has keyboard focus. Event dispatch snapshot: TextField, identifier: 'two'
[12:39:53]: ▸ TestFastlaneKeyboardIssueUITests.TestFastlaneKeyboardIssueUITests
[12:39:53]: ▸   testExample, Failed to synthesize event: Neither element nor any descendant has keyboard focus. Event dispatch snapshot: TextField, identifier: 'two'
[12:39:53]: ▸   TestFastlaneKeyboardIssueUITests.swift:40
[12:39:53]: ▸ 	 Executed 1 test, with 1 failure (0 unexpected) in 6.958 (6.959) seconds
[12:39:53]: ▸ 
[12:39:56]: ▸ 2020-03-07 12:39:56.548 xcodebuild[90166:2910066] [MT] IDETestOperationsObserverDebug: 17.499 elapsed -- Testing started completed.
[12:39:56]: ▸ 2020-03-07 12:39:56.548 xcodebuild[90166:2910066] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
[12:39:56]: ▸ 2020-03-07 12:39:56.548 xcodebuild[90166:2910066] [MT] IDETestOperationsObserverDebug: 17.499 sec, +17.499 sec -- end
[12:39:57]: ▸ Failing tests:
[12:39:57]: ▸ 	TestFastlaneKeyboardIssueUITests:
[12:39:57]: ▸ 		TestFastlaneKeyboardIssueUITests.testExample()
[12:39:57]: ▸ ** TEST FAILED **

**Comment by AndreYonadam** [comment]
I am experiencing the same issue. I tried adding Scheme's Test "Pre-actions" but had no luck.

**Comment by AndreYonadam** [comment]
I tried running the script on [these lines](https://github.com/fastlane/fastlane/blob/b677d6b4c1f89ce472edff7a87cc2b3e031b71ee/snapshot/lib/snapshot/fixes/hardware_keyboard_fix.rb#L16-L23) manually in a shell script and it turns out line 5 failed because I don't have PlistBuddy installed on my system.
